### ISSUE TITLE
RemoveFabric command crashes server when removing accessing fabric

### ISSUE THREAD

**Issue by bzbarsky-apple** [project-chip/connectedhomeip#16748]
#### Problem
After #16098 we crash if we RemoveFabric and pass it the fabric index of the fabric the command is running on.

This happens because of this code in `SessionManager::PrepareMessage`:
```
        if (session->GetSecureSessionType() == SecureSession::Type::kCASE)
        {
            FabricInfo * fabric = mFabricTable->FindFabricWithIndex(session->GetFabricIndex());
            VerifyOrDie(fabric != nullptr);
            CryptoContext::BuildNonce(nonce, packetHeader.GetSecurityFlags(), messageCounter, fabric->GetNodeId());
```
when sending the response message for the RemoveFabric command.  That `VerifyOrDie` dies, because `fabric` is in fact null by this point.

 #### Proposed Solution
We need to either fix this code to get the node id from somewhere else (e.g. store it in the session instead of getting it from the fabric table every time) or change how exactly RemoveFabric works so that it manages to send the message before doing the actual fabric removal.... but then we may have situations where we claim to have removed the fabric when we actually have not.

@kghost @tcarmelveilleux @turon

**Comment by turon** [comment]
I'd lean toward storing the node id redundantly in the session context.  That way we can support this "lingering session" use cases elegantly and decouple them from configuration updates that are occurring over the session.

**Comment by kghost** [comment]
Discussed with Boris, we will achieve `RemoveFabric` in following actions:

1. Call OnFabricDeletedFromStorage
1. Send RemoveFabric response
1. Remove session associated with the fabric
1. Remove exchanges associated with the fabric
1. Call exchange->OnExchangeReleased to all exchange delegates
1. Remove the fabric object on next schedule (or next cold boot)

**Comment by mlepage-google** [comment]
I can verify I hit this too while testing something. :-)

**Comment by Damian-Nordic** [comment]
@kghost I agree with your solution, but I think you will need to make sure that the LEAVE event is generated prior to removing ACLs for the fabric. Otherwise, generating the report containing the LEAVE event will fail the permission check, right?

**Comment by bzbarsky-apple** [comment]
Fixed in #17815.
